src/gui/menus/PlayerMenu.cpp

```diff
@@ -54,10 +68,16 @@ static void exitPlayback(void)
 	thread = NULL;
 }
 
-static int changeFile(const std::string &filepath, playbackInfo_t* playbackInfo)
+int PlayerMenu::changeFile(const std::string &filepath, playbackInfo_t* playbackInfo)
 {
 	s32 prio;
 
+	// if we don't support the file, we won't go any further
+	// to prevent unnecessarily stopping the music
+	std::string extension = filepath.substr(filepath.find_last_of('.') + 1);
+	if (!is_extension_supported(extension))
```

Comment:
This should only be done for opening non-playlist files.
m3u, pls, and json should not be in SUPPORTED_EXTS.